sticker/src/depparse/post_processing.rs

```diff
@@ -29,6 +31,48 @@ where
     }
 }
 
+/// Break cycles in the graph.
+///
+/// Panics when a token does not have a head.
+pub fn break_cycles(sent: &mut Sentence, root_idx: usize) {
+    loop {
+        let components = {
+            let digraph: &DiGraph<_, _> = (&*sent).into();
+            tarjan_scc(digraph)
+                .into_iter()
+                .filter(|c| c.len() > 1)
+                .collect::<Vec<_>>()
+        };
+
+        // We are done if there are no more cycles.
+        if components.is_empty() {
+            break;
+        }
+
+        for cycle in components.into_iter().filter(|c| c.len() > 1) {
```

Comment:
`filter` seems redundant to me, line 43 contains the same condition and is executed in the same loop iteration.

Reply:
Good catch.